lib/honeybadger/plugins/sidekiq.rb

```diff
@@ -25,20 +25,30 @@ def call(worker, msg, queue)
             ::Sidekiq.configure_server do |sidekiq|
               sidekiq.error_handlers << lambda {|ex, params|
                 job = params[:job] || params
-                retry_count = job['retry_count'.freeze].to_i
-                retry_opt = job['retry'.freeze]
-                max_retries = if retry_opt.is_a?(Integer)
-                  [retry_opt - 1, config[:'sidekiq.attempt_threshold'].to_i].min
-                else
-                  config[:'sidekiq.attempt_threshold'].to_i
+                job_retry = job['retry'.freeze]
+
+                if (threshold = config[:'sidekiq.attempt_threshold'].to_i) > 0 && job_retry
+                  # We calculate the job attempts to determine the need to
+                  # skip. Sidekiq's first job execution will have nil for the
+                  # 'retry_count' job key. The first retry will have 0 set for
+                  # the 'retry_count' key, incrementing on each execution
+                  # afterwards.
+                  retry_count = job['retry_count'.freeze]
+                  attempt = retry_count ? retry_count + 1 : 0
```

Comment:
This took me 20 min to work through in my head before I got it straight. For those coming after me who might read this here is how I reasoned it. 
```
# retry_count = number of times the job has retried
#    run 1 - retry_count is nil
#    run 2 - this is your first retry so retry_count is 0
#    run 3 - you have retried once, retry_count is 1
# attempts = number of times the job has run 
```

Reply:
Heh, I made an attempt to explain it in the comments, but your run list is much clearer.

Thanks for giving it a look @mstruve!